src/aws-cpp-sdk-core/include/aws/core/client/ClientConfiguration.h

```diff
@@ -88,6 +88,19 @@ namespace Aws
             bool shouldDisableIMDS = false;
         };
 
+        /**
+         * Configuration structure for credential providers in the AWS SDK.
+         * This structure allows passing configuration options to credential providers
+         * such as profile name and client configuration for HTTP requests made by
+         * credential providers that need to make network calls (e.g., InstanceProfileCredentialsProvider).
+         */
+        struct CredentialProviderConfiguration {
+          /**
+           * AWS profile name to use for credentials.
+           */
+          Aws::String profile;
```

Comment:
this needs to be initialized to the same value of client configuration profile in its constructor, i think we're missing that